nio/client/async_client.py

```diff
@@ -1386,11 +1385,19 @@ async def sync_forever(
             except asyncio.CancelledError:  # noqa: PERF203
                 for task in tasks:
                     task.cancel()
-
+                self._stop_sync_forever = False
+                raise
+            except:
+                self._stop_sync_forever = False
```

Comment:
Minor change, but gets the job done nicely IMO!

```suggestion
                raise
            finally:
                self._stop_sync_forever = False
```

Reply:
The finally block is always executed, resetting the flag before the next while condition check.

Reply:
Ahhh you're right, oops